### PR TITLE
POC - Use Symfony cache

### LINKED ISSUE #663: Remove Sonata Cache
We should use symfony cache instead.

Even the doctrine cache was just deprecated in favor of symfony cache:
https://github.com/doctrine/DoctrineCacheBundle#deprecation-warning

### PR DESCRIPTION
**Pull request by VincentLanglet**
## Subject

I am targeting this branch, because {reason}.

Closes #663.

## Changelog

```markdown
### Added
- Support for Symfony Cache

### Deprecated
- Support for Sonata Cache
```

### CODE REVIEW

File: src/DependencyInjection/SonataBlockExtension.php [hunk 1]
```diff
@@ -69,6 +69,11 @@ public function load(array $configs, ContainerBuilder $container): void
         $loader->load('exception.xml');
         $loader->load('commands.xml');
 
+        // NEXT_MAJOR: Remove this check and the legacy_core.xml file.
+        if (class_exists(\Sonata\Cache\CacheManagerInterface::class)) {
```
Comment (VincentLanglet) [inline src/DependencyInjection/SonataBlockExtension.php:73]: I added this in order to keep the old behavior by overriding the service if the user installed SonataCache.

File: src/Resources/config/core.xml [hunk 2]
```diff
@@ -31,7 +31,7 @@
             <argument type="service" id="sonata.block.renderer"/>
             <argument type="service" id="sonata.block.context_manager"/>
             <argument type="service" id="event_dispatcher"/>
-            <argument type="service" id="sonata.cache.manager" on-invalid="ignore"/>
+            <argument type="service" id="TODO" on-invalid="ignore"/>
```
Comment (VincentLanglet) [inline src/Resources/config/core.xml:34]: I dunno what to use as id for the symfony cache, help @sonata-project/contributors

File: src/Templating/Helper/BlockHelper.php [hunk 3]
```diff
@@ -114,23 +117,26 @@ class BlockHelper
...
+        $cacheItemPool = null,
         ?HttpCacheHandlerInterface $cacheHandler = null,
         ?Stopwatch $stopwatch = null
     ) {
         $this->blockServiceManager = $blockServiceManager;
         $this->cacheBlocks = $cacheBlocks;
         $this->blockRenderer = $blockRenderer;
         $this->eventDispatcher = $eventDispatcher;
-        $this->cacheManager = $cacheManager;
+        $this->cacheItemPool = $cacheItemPool;
```
Comment (VincentLanglet) [inline src/Templating/Helper/BlockHelper.php:139]: I need to add a deprecation here.

File: src/Resources/config/core.xml (hunk 2 above)
Comment (wbloszyk) [reply src/Resources/config/core.xml:34]: sonata.symfony_cache

Comment (VincentLanglet) [reply src/Resources/config/core.xml:34]: I don't think that's right, sonata shouldn't be in the id I'm looking for.
Event dispatcher from symfony has id `event_dispatcher`.
Here, I'm looking for the symfony service implementing the `CacheItemPoolInterface` interface.

Comment (VincentLanglet) [reply src/Resources/config/core.xml:34]: I also need to replace the `sonata.cache.noop` occurences.
https://github.com/sonata-project/SonataBlockBundle/search?q=sonata.cache.noop

Comment (wbloszyk) [reply src/Resources/config/core.xml:34]: My bad. In this place we should inject service by alias. Then alias can be configure in files core/legacy_core. Also deprecation can be easy add in legacy_core. In 5.x corrrct service can be inject directly. Then `sonata.cache_manager_alias` can be used

**Comment by VincentLanglet** [comment]
I think I made a mistake.
`CacheManagerInterface` cannot be replaced by CacheItemPoolInterface, but more by something like container, which return cache services.

File: src/Resources/config/core.xml (hunk 2 above)
Comment (VincentLanglet) [reply src/Resources/config/core.xml:34]: I think I made a huge mistake thinking CacheManagerInterface could be replaced by CacheItemPoolInterface.
But a CacheManagerInterface returns CacheServices, and CacheServices can be replaced by CacheItemPoolInterface.

So the cache_manager_alias, should be something like the container in the symfony case. This way with a service id, it returns the cache service with this id. But still I don't know what's the "recorder" in this case and how to deal with extra_cache_keys...

**Pull request closed without merging**